obspy/core/util/base.py

```diff
@@ -646,6 +647,9 @@ def download_to_file(url, filename_or_buffer, chunk_size=1024):
 
 def _generic_reader(pathname_or_url=None, callback_func=None,
                     **kwargs):
+    # convert pathlib.Path objects to str for compatibility.
+    if isinstance(pathname_or_url, Path):
+        pathname_or_url = str(pathname_or_url)
```

Comment:
Do people actually use `PurePath`? I don't really know why they have that parent class to `Path` but since you are converting to string and then file opening is done outside of pathlib, we might just make this check go for `PurePath`, I guess?

Reply:
You are right, `PurePath` should be used here. If someone wanted to simulate path handling not native to the machine they are working on they have to reach for something besides `Path` (eg `WindowsPath` can be used on linux). However, in order for that to work we would need to wrap any instance of `PurePath` in a normal `Path` to make sure the string conversion will be in the right form.

Reply:
My understanding from reading the docs is, `PurePath` is parent to `Path` implementing everything besides actual file opening. It's parent to all other Path flavors, so I would expect this is exactly what it's there for.